File: BEmu/BEmu/HistoricalDataRequest/MessageHistoric.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BEmu.HistoricalDataRequest
{
    internal class MessageHistoric : Message
    {
        private readonly ElementHistoricSecurityData _security;

        internal MessageHistoric(CorrelationID corr, string securityName, Dictionary<DateTime, Dictionary<string, object>> fieldData, int sequenceNumber) : base(new Name("TBD"), corr)
        {
            this._security = new ElementHistoricSecurityData(securityName, fieldData, sequenceNumber);
        }

        public override IEnumerable<Element> Elements { get { yield return this._security; } }
        public override object this[string name, int index] { get { return null; } }
        public override BEmu.Element AsElement { get { return new ElementReference(this); } }

        public override string ToString()
        {
            StringBuilder result = new StringBuilder();
            result.Append("HistoricalDataResponse (choice) = {" + Environment.NewLine);
            result.Append(this._security.PrettyPrint(1));
            result.Append("}");

            return result.ToString();
        }

    }
}

File: BEmuJava/src/BEmu/HistoricalDataRequest/MessageHistoric.java
package BEmu.HistoricalDataRequest;

import BEmu.Element;
import BEmu.Name;
import BEmu.Message;
import BEmu.CorrelationID;
import BEmu.Datetime;
import java.util.Map;

public class MessageHistoric extends Message
{
    private final ElementHistoricSecurityData _security;
    
    MessageHistoric(CorrelationID corr, String securityName, Map<Datetime, Map<String, Object>> fieldData, int sequenceNumber)
    {
    	super(new Name("HistoricalDataResponse"), corr, null);
    	this._security = new ElementHistoricSecurityData(securityName, fieldData, sequenceNumber);
    }
    
    ElementHistoricSecurityData security()
    {
    	return this._security;
    }
    
	public Element getElement(String name) throws Exception
	{
		if(name.toLowerCase().equals(this._security.name().toString().toLowerCase()))
			return this._security;
		else
			throw new Exception("not implemented");
	}
    
    public String toString()
    {
        StringBuilder result = new StringBuilder();
        
        result.append("HistoricalDataResponse (choice) = {" + System.getProperty("line.separator"));
        result.append(this._security.prettyPrint(1));
        result.append("}");

        return result.toString();
    }
}
